app.js

```diff
@@ -199,7 +201,37 @@ d.run(() => {
 				if (!appConfig.nethash) {
 					throw Error('Failed to assign nethash from genesis block');
 				}
-				cb(null, appConfig);
+
+				// In case domain names are used, resolve those to IP addresses.
```

Comment:
How often are those names resolved? Once on startup or periodically?

Reply:
@fchavant Only on startup; it's basically a preprocessing step done to the config file before it's used by the node. The internal logic only deals knows about IP addresses.

As part of the discovery process, peers still need to report their IPv4 addresses (we don't want peers to report themselves by domain name). If we allow peers to report anything (domain name, IPv4 or IPv6 addresses), our life will get a lot more difficult. Potentially, we could stop relying on the ip address as an identifier, but that change would be a lot more significant and I'm not sure if that would be a good idea.

Reply:
OK, that's fine for me.